### ISSUE TITLE
Unsent message trapped in app

### ISSUE THREAD

**Issue by MaryMagda** [fetlife/android#264]
Hi,

I tried to send a message yesterday from Version 2.1.3 and it seems to be trapped as an unsent message.

When I tried to send the message the red icon just kept spinning and now the message appears grey in the chat thread. When I log in to Fetlife in a browser the unsent message doesn't appear in the thread at all.

The timestamp on the message is Jan 21, 10:12:25pm and I'm running Android 6.0

Thanks!

**Comment by kistitan** [comment]
Hi @MaryMagda 

If your message cannot be sent for some reason temporary it is kept as a pending message (grayed).
As it does usually happens when you have no connection in earlier version we are waiting for the Android OS to notify us something changed with the internet connection, or another message is about to be sent.
That indeed caused some issues as in rare cases in could happen that even with living internet connection the message send failed and as there was no notification from the OS (like hey there is a living connection now) we did not try to send again.

In later version (from 2.2) we added an extra check to retry to send pending messages.

To see if you are facing the same issue, please either
- try to send an other message to anybody else with a stable connection and see if this will trigger the sending of your stucked message
- start the app and turn off and on your internet connection (wifi or 3/4g) on your device

Hopefully after any of these your stucked message will be sent. 
Please let us know the result.

**Comment by Holsann** [comment]
I am having the same issue. Version 2.5.0. Every single message I send is gray and stuck in conversation purgatory, with stable Internet connection. Doesn't matter if it's a data connection, wifi, if I restart the app or turn on and off. No messages go through at all.

**Comment by MaryMagda** [comment]
@kistitan The message did eventually get sent. I'm not quite sure what worked. I know it took more than day and I did send other messages and exit and re-enter the app a number of times.

I agree with Holsann that this seemed to be happening when other apps were able to connect and communicate.

This happened to me again just the other day and I decided to uninstall/reinstall the app to clear the draft completely rather than have it stuck in limbo.
